pkg/e2e/verify/check_workloads.go

```diff
@@ -0,0 +1,133 @@
+package verify
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/osde2e/pkg/common/alert"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+var onNodesTestName string = "[Suite: informing] [OSD] pod validating webhook"
+
+func init() {
+	alert.RegisterGinkgoAlert(onNodesTestName, "SD-SREP", "Matt Bargenquast", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+var _ = ginkgo.Describe(onNodesTestName, func() {
+
+	h := helper.New()
+
+	ginkgo.Context("worker nodes", func() {
+
+		ginkgo.It("on worker nodes", func() {
+
+			_, infra, err := checkWorkloadsOnWorkerNodes(h.Kube().CoreV1())
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = checkPods(h.Kube().CoreV1(), infra)
+			Expect(err).NotTo(HaveOccurred())
+
+		})
+	})
+})
+
+func checkPods(podClient v1.CoreV1Interface, infra []string) (map[string]string, error) {
+
+	violators := make(map[string]string)
+
+	operators := map[string]string{"cloud-ingress-operator": "openshift-cloud-ingress-operator",
+		"configure-alertmanager-operator": "openshift-monitoring",
+		"custom-domains-operator":         "openshift-custom-domains-operator",
+		"managed-upgrade-operator":        "openshift-managed-upgrade-operator",
+		"managed-velero-operator":         "openshift-velero",
+		"must-gather-operator":            "openshift-must-gather-operator",
+		"osd-metrics-exporter":            "openshift-osd-metrics",
+		"rbac-permissions-operator":       "openshift-rbac-permissions"}
+
+	listOpts := metav1.ListOptions{}
+	list, err := podClient.Pods(metav1.NamespaceAll).List(context.TODO(), listOpts)
```

Comment:
Given that there may be many pods, do we need to handle pagination here?

Reply:
Right now the code runs through this bit very fast. I don't think that the number of pods will reach such huge numbers that it will make a difference.